src/test/java/redis/clients/jedis/util/JedisURIHelperTest.java

```diff
@@ -57,4 +60,21 @@ public void shouldValidateInvalidURIs() throws URISyntaxException {
     assertFalse(JedisURIHelper.isValid(new URI("redis://host/0")));
   }
 
+  @Test
+  public void shouldGetDefaultProtocolWhenNotDefined() {
+    assertNull(getRedisProtocol(URI.create("redis://host:1234")));
+    assertNull(getRedisProtocol(URI.create("redis://host:1234/1")));
+  }
+
+  @Test
+  public void shouldGetProtocolFromShortDefinition() {
+    assertEquals(RedisProtocol.RESP3, getRedisProtocol(URI.create("redis://host:1234?protocol=3")));
+    assertEquals(RedisProtocol.RESP3, getRedisProtocol(URI.create("redis://host:1234/1?protocol=3")));
+  }
+
+  @Test
+  public void shouldGetProtocolFromLongDefinition() {
+    assertEquals(RedisProtocol.RESP3, getRedisProtocol(URI.create("redis://host:1234?protocol=resp3")));
```

Comment:
Note, according to the [specification](https://github.com/redis/redis-specifications/blob/bd7f8a6c52e00ee853758ec3cf213901fc2b7c98/uri/redis.txt#L106) the value of protocol should be an integer, always.